bundles/CoreBundle/Command/Migrate/StorageCommand.php

```diff
@@ -15,81 +15,104 @@
 
 namespace Pimcore\Bundle\CoreBundle\Command\Migrate;
 
-use League\Flysystem\FilesystemOperator;
 use League\Flysystem\StorageAttributes;
 use Pimcore\Console\AbstractCommand;
-use Pimcore\Tool\Storage;
+use Psr\Container\ContainerInterface;
 use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputArgument;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
 
 /**
  * @internal
  */
-class ThumbnailsFolderStructureCommand extends AbstractCommand
+class StorageCommand extends AbstractCommand
 {
+    /**
+     * @param ContainerInterface $locator
+     */
+    public function __construct(private ContainerInterface $locator)
+    {
+        parent::__construct();
+    }
+
     protected function configure()
     {
         $this
-            ->setName('pimcore:migrate:thumbnails-folder-structure')
-            ->setDescription('Change thumbnail folder structure to <asset storage>/<path to asset>/<asset id>/image-thumb__<asset id>__<thumbnail name>/<asset filename> instead of <asset storage>/<path to asset>/image-thumb__<asset id>__<thumbnail name>/<asset filename>');
+            ->setName('pimcore:migrate:storage')
+            ->setDescription('Migrate data from one storage to another')
+            ->addArgument(
+                'storage',
+                InputArgument::IS_ARRAY,
+                'A list of storages to be migrated'
+            );
     }
 
     /**
      * {@inheritdoc}
      */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $thumbnailStorage = Storage::get('thumbnail');
+        $storages = $input->getArgument('storage');
 
-        $output->writeln('Migrating thumbnails ...');
-        $this->doMigrateStorage($output, $thumbnailStorage);
-        $output->writeln("\nSuccessfully moved thumbnail files to new folder structure\n");
+        foreach ($storages as $storageName) {
+            $storageSourceName = $this->getStorageName($storageName, 'source');
+            $storageTargetName = $this->getStorageName($storageName, 'target');
 
-        $assetCacheStorage = Storage::get('asset_cache');
-
-        $output->writeln('Migrating asset cache (document previews, etc.) ...');
-        $this->doMigrateStorage($output, $assetCacheStorage);
-        $output->writeln("\nSuccessfully moved asset cache files to new folder structure");
-
-        return 0;
-    }
+            try {
+                $sourceStorage = $this->locator->get($storageSourceName);
+                $targetStorage = $this->locator->get($storageTargetName);
```

Comment:
@fashxp This means that you have to configure `pimcore.thumbnail.storage.source` and `target`, otherwise you cannot profit of the new directory structure. Is this really necessary or should we fall back to same storage location? Because usually those services will not have been set if the system never siwtched thumbnail storage.

Reply:
@BlackbitNeueMedien No, look at the command name, it's about the existing storage migration command, see https://pimcore.com/docs/pimcore/current/Development_Documentation/Installation_and_Upgrade/System_Setup_and_Hosting/File_Storage_Setup.html#page_Storage-Migration not about the thumbnail structure 😉
The Thumbnail structure can be migrated using the `ThumbnailsFolderStructureCommand`.

Reply:
Ah sorry, thanks for explanation.
Rest of @fashxp's changes look good to me.